src/test/fuzz/addrman.cpp

```diff
@@ -25,25 +26,208 @@ void initialize_addrman()
+    {
+        WITH_LOCK(cs, insecure_rand = FastRandomContext{ConsumeUInt256(fuzzed_data_provider)});
+        if (fuzzed_data_provider.ConsumeBool()) {
+            m_asmap = ConsumeRandomLengthBitVector(fuzzed_data_provider);
+            if (!SanityCheckASMap(m_asmap)) {
+                m_asmap.clear();
+            }
+        }
+    }
+
+    /**
+     * Generate a random address. Always returns a valid address.
+     */
+    CNetAddr RandAddr() EXCLUSIVE_LOCKS_REQUIRED(cs)
+    {
+        CNetAddr addr;
+        if (m_fuzzed_data_provider.remaining_bytes() > 1 && m_fuzzed_data_provider.ConsumeBool()) {
+            addr = ConsumeNetAddr(m_fuzzed_data_provider);
+        } else {
+            // The networks [1..6] correspond to CNetAddr::BIP155Network (private).
+            static const std::map<uint8_t, uint8_t> net_len_map = {{1, ADDR_IPV4_SIZE},
+                                                                   {2, ADDR_IPV6_SIZE},
+                                                                   {4, ADDR_TORV3_SIZE},
+                                                                   {5, ADDR_I2P_SIZE},
+                                                                   {6, ADDR_CJDNS_SIZE}};
+            uint8_t net = insecure_rand.randrange(5) + 1; // [1..5]
+            if (net == 3) {
+                net = 6;
+            }
+
+            CDataStream s(SER_NETWORK, PROTOCOL_VERSION | ADDRV2_FORMAT);
+
+            s << net;
+            s << insecure_rand.randbytes(net_len_map.at(net));
+
+            s >> addr;
+        }
+
+        // Return a dummy IPv4 5.5.5.5 if we generated an invalid address.
+        if (!addr.IsValid()) {
+            in_addr v4_addr = {};
+            v4_addr.s_addr = 0x05050505;
+            addr = CNetAddr{v4_addr};
+        }
+
+        return addr;
+    }
+
+    /**
+     * Fill this addrman with lots of addresses from lots of sources.
+     */
+    void Fill()
     {
-        WITH_LOCK(cs, insecure_rand = FastRandomContext{random_seed});
-        Clear();
+        LOCK(cs);
+
+        // Add some of the addresses directly to the "tried" table.
+
+        // 0, 1, 2, 3 corresponding to 0%, 100%, 50%, 33%
+        const size_t n = m_fuzzed_data_provider.ConsumeIntegralInRange<size_t>(0, 3);
+
+        const size_t num_sources = m_fuzzed_data_provider.ConsumeIntegralInRange<size_t>(10, 50);
+        CNetAddr prev_source;
+        // Use insecure_rand inside the loops instead of m_fuzzed_data_provider because when
+        // the latter is exhausted it just returns 0.
+        for (size_t i = 0; i < num_sources; ++i) {
+            const auto source = RandAddr();
+            const size_t num_addresses = insecure_rand.randrange(500) + 1; // [1..500]
+
+            for (size_t j = 0; j < num_addresses; ++j) {
+                const auto addr = CAddress{CService{RandAddr(), 8333}, NODE_NETWORK};
+                const auto time_penalty = insecure_rand.randrange(100000001);
+#if 1
```

Comment:
What is this for?

Reply:
Change it to `#if 0` to enable the equivalent `#else` branch which is much shorter and clear, but also slower. The `#else` snippet also shows what the faster snippet is trying to achieve by "manually" fiddling with addrman internals.